pkg/operator/sync.go

```diff
@@ -1240,6 +1241,31 @@ func (optr *Operator) reconcileMachineOSBuilder(mob *appsv1.Deployment) error {
 		return fmt.Errorf("could not get layered MachineConfigPools: %w", err)
 	}
 
+	// If layeredMCP is found, verify that the node is coreos based as non coreos OS images
+	// cannot handle layered builds
+	if len(layeredMCPs) > 0 {
+		nodes, err := optr.GetAllManagedNodes(layeredMCPs)
```

Comment:
suggestion (non-blocking): Let's move everything in the `if len(layeredMCPs) {}` block into a separate method on the `Operator` struct. We could call it something like: `optr.validateLayeredPoolNodes()`.

Reply:
Fixed - created a new method and moved the logic there.